subsys/logging/log_core.c

```diff
@@ -62,11 +64,121 @@ static k_tid_t proc_tid;
 static u32_t dummy_timestamp(void);
 static timestamp_get_t timestamp_func = dummy_timestamp;
 
+
+bool log_is_strdup(const void *buf);
+
 static u32_t dummy_timestamp(void)
 {
 	return 0;
 }
 
+/**
+ * @brief Count number of string format specifiers (%s).
+ *
+ * Result is stored as the mask (argument n is n'th bit). Bit is set if %s was
+ * found.
+ *
+ * @param str String.
+ * @param nargs Number of arguments in the string.
+ *
+ * @return Mask with %s format specifiers found.
+ */
+static u32_t count_s(const char *str, u32_t nargs)
+{
+	char curr = *str++;
+	bool arm = (curr == '%');
+	u32_t arg = 0;
+	u32_t mask = 0;
+	__ASSERT_NO_MSG(nargs <= 8*sizeof(mask));
+
+	while (curr) {
```

Comment:
1. Scanning is done inside loop and in variable declaration. Could we move it into single place?
2. You do not handle optional flags (#, 0-9, -, ' ').

Reply:
no. 1 done
no. 2 I would prefer to not do it. it increases complexity of the scanner significantly and i don't see those being used in the logger messages. Prefer to wait for someone requesting that. Then it could be added as option if needed.

Reply:
This leaves grey area which could lead to errors (undetected lack of log_strdup() usage)

Reply:
i found bug in current algorithm, need to fix that. I will see if im able to add those as well ... but i no promise.

Reply:
Algorithm, fixed and test added @pizi-nordic can you take a look once again?